### DISCUSSION TITLE
It's extremely difficult to install the library

### DISCUSSION THREAD

**Discussion by 404-html** [tamagui/tamagui#1125] [General]
### Current Behavior

We had an attempt to add tamagui library to both: react native and next js projects.
Both unfortunately unsuccessful. 

We followed the documentation. Also tried different things, e.g.:
[https://github.com/tamagui/tamagui/issues/524#issuecomment-1398576145](https://github.com/tamagui/tamagui/issues/524#issuecomment-1398576145)

[https://github.com/tamagui/tamagui/issues/303#issuecomment-1319008879](https://github.com/tamagui/tamagui/issues/303#issuecomment-1319008879)

Still no luck with getting the library up and running.
Similar issues were already raised - and closed with no conclusion. So I'm raising a new one wondering if anything can be done to make tamagui installation process smoother. 

### Expected Behavior

Better installation experience. I'm sure this great library can get more attention if this will be improved.

### Tamagui Version

```markdown
1.23.0
```

### Reproduction

```markdown
Follow the steps from documentation.
```

**Comment by natew** [comment]
This morning I released a new starter template that shows off a complete configuration from scratch, plus sets up webpack and vite.

I'll leave this open and in the future will run through the entire docs again and see.

It would be super helpful for you to do that and push it to a repo somewhere for me to debug. This is OSS and we have tons of resources out there, many starter templates, etc. If you are stuck following docs you should grab any of the starters and go from there. But it's valid if there is some missing step.

**Reply by KamilDike** [reply]
I've tried a bunch of setups. Here is [attempt](https://github.com/KamilDike/tamagui-repro) and it's been discussed at discord as well https://discord.com/channels/909986013848412191/1101536581715775638

**Comment by 404-html** [comment]
What could potentially simplify the installation process is getting rid of Linear Gradients step, as it requires Expo installation and it's not used it every project. I'm not sure how crucial this is for Tamagui and what alternatives are.

I did a quick search and found following:
https://github.com/react-native-linear-gradient/react-native-linear-gradient

I believe that [React Native Skia](https://github.com/Shopify/react-native-skia) can handle gradients quite well but it's still in beta stage and it's adding quite a lot to the bundle size.

**Reply by natew** [reply]
It just requires the core expo module, which is as easy and unobtrusive as any other native dep (correct me if I'm wrong). It also isn't included by default anymore, only when you explicitly import `tamagui/linear-gradient`.

**Comment by mkurczewski** [comment]
I also tried few attempts to install this lib but... it's definitely not meant for me. I'm using NX and tried to put the `TAMAGUI_TARGET` env everywhere and still getting the error for iOS:
```
'No process.env.TAMAGUI_TARGET set, please set it to "native" or "web".'
```

On Android on the other side, I'm getting 
```
WARN  No font size found $true undefined in size tokens ["$4"]
```

Effect is that for both platforms the app freezes on bootsplash 😢 and the only thing I wan't to render is a component with one Button:

```TSX
// app.tsx
const App = () => {
  return <Button>Test</Button>
}

// main.tsx
const Main = () => (
  <TamaguiProvider config={config}>
    <Suspense fallback={<Text>Loading... </Text>}>
      <AppPerf />
    </Suspense>
  </TamaguiProvider>
)

AppRegistry.registerComponent("main", () => Main)
```

I'm not using Expo or Vite, just vanilla react-native project with metro & babel configured by Nx. This has for example an implication that because of the babel configuration being placed in `json` file, not `js` one, I can't add `process.env.TAMAGUI_TARGET = "native"` here (and even after changing it to `js` and adding mentioned line, the error persist). 

But in fact, such kind of setup should be done a different way than setting the env. For example, why can't it be just another parameter in babel/config file like:

```TSX
    [
      "@tamagui/babel-plugin",
      {
        components: ["tamagui"],
        config: "./src/tamagui.config.ts",
        target: "native"
      },
    ],
```
and then let people decide whether they want to use a variable here or a static value?

In general, it is clear to me that this project is definitely not universal and I don't like it but understand. However, it'd be nice if you could just list libraries/frameworks that your library is fully compatible with, along with some basic examples in the repo. It would be much clearer what we can expect to be fully working and what is "experimental".

**Reply by 404-html** [reply]
> But in fact, such kind of setup should be done a different way than setting the env. For example, why can't it be just another parameter in babel/config file like:

Neat idea! I was also struggling with setting `TAMAGUI_TARGET`. It all looked like it was set but the library was still behaving like it's not: https://github.com/tamagui/tamagui/issues/1185

> However, it'd be nice if you could just list libraries/frameworks that your library is fully compatible with, along with some basic examples in the repo.

I believe clarifying that it's a process and there's a huge number of different setups o be tested. I think the author of the library have a vision and supporting all the cases could kill it. I understand that, although I'm not happy with my issue being closed 😄  It's an open source project so we can try and contribute.

**Reply by natew** [reply]
We do list the supported configurations all over the site. Vite, webpack, expo, babel, all well supported. Yarn and npm generally for package managers.

Nx has given people trouble but there are some community examples getting it working I think in the community section.

If we don't have it in the docs its not officially supported, we can't list every config we don't support.
